[R2] Document writeHttpMetadata method on R2Object (#4452)
cloudflare/cloudflare-docs: content/r2/runtime-apis.md

  - A map of custom, user-defined metadata associated with the object.

- {{<code>}}writeHttpMetadata(headers{{<param-type>}}Headers{{</param-type>}}){{</code>}} {{<type>}}void{{</type>}}

>   -  Retrieves the `httpMetadata` from the R2Object and writes to a Headers object. See [HTTP Metadata](#http-metadata).

Reviewer: ```suggestion
  -  Retrieves the `httpMetadata` from the R2Object and writes to a Headers object. Refer to [HTTP Metadata](#http-metadata).
```

>   -  Retrieves the `httpMetadata` from the R2Object and writes to a Headers object. Refer to [HTTP Metadata](#http-metadata).

Reviewer: ```suggestion
  -  Retrieves the `httpMetadata` from the `R2Object` and writes to a `Headers` object. Refer to [HTTP Metadata](#http-metadata).
```

>   -  Retrieves the `httpMetadata` from the `R2Object` and writes to a `Headers` object. Refer to [HTTP Metadata](#http-metadata).

Reviewer 2: ```suggestion
  -  Retrieves the `httpMetadata` from the `R2Object` and applies their corresponding HTTP headers to the `Headers` input object. Refer to [HTTP Metadata](#http-metadata).
```
